content/kubermatic/master/upgrading/2.14_to_2.15/_index.en.md

```diff
@@ -5,7 +5,384 @@ weight = 90
+If you do not provide a kubeconfig, Seeds will still be converted, but you are responsible
+yourself to provide the appropriate kubeconfig Secrets and to reference them in the Seeds.
+
+```bash
+./kubermatic-installer convert-datacenters --pause-seeds --kubeconfig kubeconfig-with-all-seeds datacenters.yaml
+# apiVersion: kubermatic.k8s.io/v1
+# kind: Seed
+# ...
+#
+# ---
+# ...
+```
+
+Refer to the [KubermaticConfiguration]({{< ref "../../concepts/kubermaticconfiguration" >}}) and
+[Seed]({{< ref "../../concepts/seeds" >}}) documentation for more information.
+
+Once the conversion is completed, carefully check the new configuration files for mistakes. Do
+note that the installer generally does not output default values, so if you configured for example
+2 Apiserver replicas and this is the default for Kubermatic anyway, the generated `KubermaticConfiguration`
+will skip the value alltogether, relying on the defaulting.
+
+It is now recommended to setup Kubermatic in a test environment, using the newly converted files.
+
+#### Upgrading the Master Cluster
+
+It's now time to change the Helm charts. Begin by uninstalling the `kubermatic` chart. This will make
+the dashboard unavailable and begin the downtime.
+
+**Helm 3**
+
+```bash
+helm --namespace kubermatic delete kubermatic
+```
+
+**Helm 2**
+
+```bash
+helm --tiller-namespace kubermatic-installer delete --purge kubermatic
+```
+
+Once the Helm chart is gone, it's time to perform a clean installation of the new Kubermatic Operator.
+The first step for that is to add new Kubermatic CRDs:
+
+```bash
+kubectl apply -f charts/kubematic/crd/
+```
+
+To prevent Kubermatic from rejecting existing clusters, make sure to apply the Seeds before installing
+the Operator. We assume that `seeds-with-secrets.yaml` is part of the output of one of the conversion
+commands described above.
+
+```bash
+kubectl apply -f seeds-with-secrets.yaml
+```
+
+Check that all seeds are in place now:
+
+```bash
+kubectl -n kubermatic get seeds
+```
+
+Now it's time to install the operator. Make sure to configure the `kubermatic-operator` chart properly,
+taking care of setting the correct ImagePullSecret in the `values.yaml`. Once you're satisfied, install
+the operator:
+
+**Helm 3**
+
+```bash
+helm --namespace kubermatic install --values myvalues.yaml kubermatic-operator charts/kubermatic-operator/
+```
+
+**Helm 2**
+
+```bash
+helm --tiller-namespace kubermatic-installer upgrade --install --values myvalues.yaml --namespace kubermatic kubermatic-operator charts/kubermatic-operator/
+```
+
+Finally, apply the KubermaticConfiguration. This will make the operator begin reconciling the master
+cluster. Once again, we assume that this config file has been produced by the `convert-helm-values`
+command:
+
+```bash
+kubectl apply -f kubermaticconfiguration.yaml
+```
+
+It will now take a few minutes for the new Kubermatic installation to become ready.
+
+#### Migrate Seeds
```

Comment:
At this point we have master-controller-manager and seed-controller-managers with misaligned versions, this means that creating a cluster at this point could end-up with a failure. Should we somehow disable cluster creations or prompt a warning in the UI?

Reply:
I'm not quite sure how to handle this. I think a notice in the dashboard would be too much and also be maybe a bit too late for the 2.15 release (not sure how much effort it is to introduce such a sticky note in the dashboard). Plus some customers use whitelabelled forks of the dashboard anyway.

I think I would prefer a more invisible stopgap. Can we prevent the cluster rollout by simply not copying the Cluster resources onto the paused seeds?

Reply:
As we discussed offline, we could just make sure that the documentation is clear on this point. If users will report issues related to this we could think of introducing some mechanisms.

> Can we prevent the cluster rollout by simply not copying the Cluster resources onto the paused seeds?

I think that doing this without any explicit notification would be cause of confusion.